src/Workspaces/CoreTestUtilities/Options/OptionsTestInfo.cs

```diff
@@ -3,20 +3,27 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using System.Collections.Generic;
+using System.Collections.Immutable;
 using System.IO;
-using System.Linq;
 using System.Reflection;
 using Microsoft.CodeAnalysis.Options;
 using Xunit;
 
 namespace Microsoft.CodeAnalysis.UnitTests;
 
-internal readonly record struct OptionsTestInfo(IOption2 Option, List<(string namespaceName, string qualifiedName, bool isPublic, IOption2 option)> Accessors)
+internal readonly record struct OptionsTestInfo(IOption2 Option, ImmutableList<(string namespaceName, string qualifiedName, bool isPublic, IOption2 option)> Accessors)
 {
-    public static Dictionary<string, OptionsTestInfo> CollectOptions(string directory)
+    private static Tuple<string, ImmutableDictionary<string, OptionsTestInfo>>? s_cachedResult;
+
+    public static ImmutableDictionary<string, OptionsTestInfo> CollectOptions(string directory)
     {
-        var result = new Dictionary<string, OptionsTestInfo>();
+        if (s_cachedResult is var (cachedResultDirectory, cachedResult)
```

Comment:
Perhaps we don't need to cache the values here, but instead in the caller. We wouldn't need to maintain per-directory data since the caller likely only uses a single directory.